File: js/complete/no029.js
#!/usr/bin/env node

var fns = require('../functions.js'),
    operator = require('../operator.js'),
    timer = require('../timer.js');

exports.main = function(verbose) {
    if (typeof verbose == 'undefined') {
        verbose = false;
    }
    var n = 100, powers = fns.applyToList(operator.bigIntPow, operator.range(2, n + 1));
    powers = operator.uniqSorted(powers, false);
    return powers.length;
};

if (require.main === module) {
    timer.timer(29, exports.main);
}

File: python/complete/no029.py
#!/usr/bin/env python

import operator

from python.decorators import euler_timer
from python.functions import apply_to_list


def main(verbose=False):
    n = 100
    powers = apply_to_list(operator.pow, range(2, n + 1))
    return len(set(powers))

if __name__ == '__main__':
    print euler_timer(29)(main)(verbose=True)
